### PR TITLE
Global model

### PR DESCRIPTION
**Pull request by ykate1998**
# Description

Adding the CNN Image Retrieval model implementation based on the following papers:

[1] Fine-tuning CNN Image Retrieval with No Human Annotation,
Radenović F., Tolias G., Chum O., TPAMI 2018 [arXiv]
https://arxiv.org/abs/1711.02512

[2] CNN Image Retrieval Learns from BoW: Unsupervised Fine-Tuning with Hard
Examples, Radenović F., Tolias G., Chum O., ECCV 2016 [arXiv]
https://arxiv.org/abs/1604.02426

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] A new research paper code implementation

### CODE REVIEW

File: research/delf/delf/python/training/model/global_model.py
Comment (andrefaraujo) [inline research/delf/delf/python/training/model/global_model.py:259]: maybe rename to extract_multi_scale_descriptor

File: research/delf/delf/python/training/model/global_model.py
Comment (andrefaraujo) [inline research/delf/delf/python/training/model/global_model.py:266]: same terminology comment as in the previous function

File: research/delf/delf/python/training/model/global_model_test.py
Comment (andrefaraujo) [inline research/delf/delf/python/training/model/global_model_test.py:72]: testExtractMultiScale

**Review by andrefaraujo — CHANGES_REQUESTED** [review]
Thanks, Kate! It looks good overall, the comments I have are generally minor. Please take a look :)

File: research/delf/delf/python/training/model/global_model.py
Comment (ykate1998) [reply research/delf/delf/python/training/model/global_model.py:114]: As far as I can tell, all models are imported in keras.Model format, not only EfficientNet. The code you pointed out seems to be working in the case of DenseNet

Comment (ykate1998) [reply research/delf/delf/python/training/model/global_model.py:119]: Sorry, you were right, I tested this specific pass and it needs the conversion to Sequential as you mentioned

**Comment by ykate1998** [comment]
@andrefaraujo , I believe it should be fixed now. Thank you for the review

File: research/delf/delf/python/training/model/global_model.py
Comment (andrefaraujo) [reply research/delf/delf/python/training/model/global_model.py:122]: Thanks!

File: research/delf/delf/python/training/model/global_model_test.py
Comment (andrefaraujo) [inline research/delf/delf/python/training/model/global_model_test.py:70]: fix indentation here (pull code to previous line)

**Review by andrefaraujo — CHANGES_REQUESTED** [review]
Thanks Kate! Looks great. Just one small change left, and I will merge the code once you make the edit.

**Comment by ykate1998** [comment]
@andrefaraujo, should be fixed now :) Thank you

**Review by andrefaraujo — APPROVED** [review]
Thanks, Kate!

**Comment by andrefaraujo** [comment]
@ykate1998 can you please resolve the conflict? I cannot merge it right now. Once you resolve the conflict, I'll be able to merge. I guess it may be related to the other PR which I just merged.

**Comment by ykate1998** [comment]
Hello, should be fixed now. Thanks

**Review by andrefaraujo — APPROVED** [review]
Thanks Kate! Merging...

**Pull request merged**
